examples/NFD/syn_flood_detection/SYNFloodDetection.cpp

```diff
@@ -0,0 +1,360 @@
+        short protocol = ntohs(*(short *)(ip_header + 9));
+        this->field_value["UDP"] = new int((protocol == IPPROTO_UDP) ? 1 : 0);
+
+        /*TCP layer*/
+        TCPHdr *tcph = (TCPHdr *)(ip_header + ip_header_length);
+        this->field_value["sport"] = new int(ntohs(tcph->th_sport));
+        this->field_value["dport"] = new int(ntohs(tcph->th_dport));
+        /*URG ACK PSH RST SYN FIN*/
+        this->field_value["flag_fin"] = new int(ntohs(tcph->th_flags) & TH_FIN);
+        this->field_value["flag_syn"] = new int(ntohs(tcph->th_flags) & TH_SYN);
+        this->field_value["flag_ack"] = new int(ntohs(tcph->th_flags) & TH_ACK);
+}
+void
+Flow::clean() {
+        /*Encoding*/
+        struct ether_header *eth_header;
+        eth_header = (struct ether_header *)this->pkt;
+        int ethernet_header_length = 14; /* Doesn't change */
+        int ip_header_length;
+
+        u_char *ip_header;
+        ip_header = (u_char *)eth_header + ethernet_header_length;
+        ip_header_length = ((*ip_header) & 0x0F);
+        ip_header_length = ip_header_length * 4;
+
+        *((int *)(ip_header + 12)) = htonl(*((int *)this->field_value["sip"]));
+        *((int *)(ip_header + 16)) = htonl(*((int *)this->field_value["dip"]));
+        // this->field_value["iplen"] = new int(totallength);
+
+        /*TCP layer*/
+        TCPHdr *tcph = (TCPHdr *)(ip_header + ip_header_length);
+        tcph->th_sport = u_short(*((int *)this->field_value["sport"]));
+        tcph->th_dport = u_short(*((int *)this->field_value["dport"]));
+}
+
+int
+process(Flow &f);
+Flow f_glb;
+long int _counter = 0;
+unordered_map<string, int> F_Type::MAP = unordered_map<string, int>();
+unordered_map<string, int> F_Type::MAP2 = unordered_map<string, int>();
+
+struct timeval begin_time;
+struct timeval end_time;
+
+void
+_init_() {
+        (new F_Type())->init();
+}
+int _t1 = 100;
+int _t2 = 1;
+int _t3 = 1;
+int _t4 = 1;
+int _t5 = 1;
+int _t6 = 1;
+int _t7 = 1;
+int _t8 = 1;
+int _t9 = 1;
+int _t10 = 1;
+int _t11 = 1;
+int _t12 = 1;
+int _t13 = 1;
+int _t14 = 1;
+State<unordered_map<IP, int>> blist(*(new unordered_map<IP, int>()));
+State<int> threshold(_t1);
+
+int
+process(Flow &f) {
+        if ((*(int *)f["flag_syn"]) == _t2 && (*(int *)f["tag"]) != _t3) {
+                blist[f][(*(IP *)f["sip"])] = blist[f][(*(IP *)f["sip"])] + _t4;
+                (*(int *)f["tag"]) = _t5;
+        } else if (((*(int *)f["tag"]) == _t6) && (blist[f][(*(IP *)f["sip"])] == threshold[f])) {
+                return -1;
+        } else if (((*(int *)f["tag"]) == _t7) && (blist[f][(*(IP *)f["sip"])] != threshold[f])) {
+        } else if ((*(int *)f["tag"]) != _t8 && (*(int *)f["flag_syn"]) != _t9 && (*(int *)f["flag_ack"]) == _t10) {
+                blist[f][(*(IP *)f["sip"])] = blist[f][(*(IP *)f["sip"])] - _t11;
+        } else if ((*(int *)f["tag"]) != _t12 && (*(int *)f["flag_syn"]) != _t13 && (*(int *)f["flag_ack"]) != _t14) {
```

Comment:
What purpose does this else if branch have

Reply:
the empty branch means that we do no operation on the packets but just pass it.